milli/src/search/mod.rs

```diff
@@ -284,20 +285,60 @@ pub fn word_derivations<'c>(
         Entry::Occupied(entry) => Ok(entry.into_mut()),
         Entry::Vacant(entry) => {
             let mut derived_words = Vec::new();
-            let dfa = build_dfa(word, max_typo, is_prefix);
-            let mut stream = fst.search_with_state(&dfa).into_stream();
+            if max_typo == 0 {
+                if is_prefix {
+                    let prefix = Str::new(word).starts_with();
+                    let mut stream = fst.search(prefix).into_stream();
+
+                    while let Some(word) = stream.next() {
+                        let word = std::str::from_utf8(word)?;
+                        derived_words.push((word.to_string(), 0));
+                    }
+                } else {
+                    let automaton = Str::new(word);
+                    let mut stream = fst.search(automaton).into_stream();
+                    while let Some(word) = stream.next() {
+                        let word = std::str::from_utf8(word)?;
+                        derived_words.push((word.to_string(), 0));
+                    }
```

Comment:
so there is no `get` on fst, but since we are looking for an  exact match, isn't that equivalent to:

```rust
                    if fst.contains(word) {
                        derived_words.push((word.to_string(), 0));
                    }

```
?